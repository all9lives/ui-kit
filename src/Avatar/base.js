import { css } from 'styled-components'
import React, { Component } from 'react'
import Unit from '../utils/Unit'
import Size from '../utils/Size'
import Color from '../utils/Color'
import View from '../View'
import Image from '../Image'
import Text from '../Text'

const SIZE_MAPPING = {
  small: 2,
  regular: 3,
  large: 4
}

export const Label = View.extend`
  position: absolute;
  border-radius: ${Unit.parse(3)};
  padding: ${Unit.parse(Size.spacing * 0.3)};
  background-color: ${Color.primary_light};
  right: ${Unit.parse(Size.spacing * 0.1)};
  bottom: ${props => props.name ? Unit.parse(Size.spacing * 2) : Unit.parse(Size.spacing * 0.1)};
`

export const LabelText = Text.extend`
  font-size: ${Unit.parse(Size.small)};
  color: ${Color.white};
`

export const Name = View.extend`
  margin-top: ${Unit.parse(Size.spacing * 0.5)};
  background-color: ${Color.gray};
  border-radius: ${Unit.parse(Size.small)};
  align-items: center;
  justify-content: center;
`

export const NameText = Text.extend`
  font-size: ${Unit.parse(Size.small)};
  color: ${Color.white};
`

export class Base extends Component {
  static defaultProps = {
    source: require('../assets/avatar-cat.jpg')
  }
  renderLabel = ({ label, name, labelStyle, labelTextStyle }) => (
    <Label name={name} style={labelStyle}>
      <LabelText style={labelTextStyle}>{label}</LabelText>
    </Label>
  )
  renderName = ({ name, nameStyle, nameTextStyle }) => (
    <Name style={nameStyle}>
      <NameText style={nameTextStyle}>{name}</NameText>
    </Name>
  )
  render () {
    const { size, source, name, label, borderRadius } = this.props
    return (
      <View direction='column' {...this.props}>
        <Image size={Size.icon * SIZE_MAPPING[size]} source={source} style={{ borderRadius }} />
        {name && this.renderName(this.props)}
        {label && this.renderLabel(this.props)}
      </View>
    )
  }
}

const base = css`
  position: relative;
`

export default base
