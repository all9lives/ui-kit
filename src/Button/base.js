import { css } from 'styled-components'
import Color from '../utils/Color'
import Unit from '../utils/Unit'
import Size from '../utils/Size'

export const attrs = {
  color: props => props.primary
    ? Color.primary
    : (Color[props.color] || props.color || Color.black)
}

const base = css`
  margin: ${props => Unit.parse(props.margin, (margin) => margin * Size.spacing)};
`

export default base
