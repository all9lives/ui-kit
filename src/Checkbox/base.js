import React, { PureComponent } from 'react'
import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Spacing from '../utils/Spacing'
import Text from '../Text'

const base = css`
  position: relative;
  margin-right: ${Unit.convert(Spacing.regular)};
`

export const getComponent = ({ View, Checkbox }) => class Component extends PureComponent {
  handleValueChange = (value) => {
    if (this.props.onValueChange) {
      this.props.onValueChange(value)
    }
  }
  handleLabelPress = () => {
    if (this.props.onValueChange) {
      this.props.onValueChange(!this.props.value)
    }
  }
  render () {
    const { value, label } = this.props
    return (
      <View align='middle'>
        <Checkbox value={value} onValueChange={this.handleValueChange} />
        <Text alignSelf='center' onPress={this.handleLabelPress}>{label}</Text>
      </View>
    )
  }
}

export default base
