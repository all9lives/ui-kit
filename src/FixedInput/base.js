import { css } from 'styled-components'
import Color from '../utils/Color'

const base = css`
  background-color: ${Color.white};
`

export default base
