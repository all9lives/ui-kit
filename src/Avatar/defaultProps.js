export default {
  size: 'regular',
  borderRadius: 0
}
