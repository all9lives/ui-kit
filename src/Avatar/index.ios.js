import Avatar from './index.android'

export default Avatar
