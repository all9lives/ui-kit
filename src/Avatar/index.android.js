import styled from 'styled-components/native'
import base, { Base } from './base'
import defaultProps from './defaultProps'

const Avatar = styled(Base)`
  ${base}
`
Avatar.defaultProps = defaultProps

export default Avatar
