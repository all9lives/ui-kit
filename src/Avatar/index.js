import styled from 'styled-components'
import base, { Base } from './base'
import defaultProps from './defaultProps'

const Avatar = styled(Base)`
  ${base}
`
Avatar.defaultProps = defaultProps

export default Avatar
