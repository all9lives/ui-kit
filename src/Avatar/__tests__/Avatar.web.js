/* eslint-env jest */
import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Avatar from '../index.js'
import { Base, Label } from '../base'

const diveToBase = wrapper => wrapper.find(Base).dive()
const diveToLabel = wrapper => wrapper.find(Base).dive().find(Label).dive()

describe('Avatar.web', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Avatar>Apple</Avatar>
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `borderRadius:5`', () => {
    const wrapper = shallow(
      <Avatar borderRadius={5} />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `name:foo`', () => {
    const wrapper = shallow(
      <Avatar name='foo' />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `label:boo`', () => {
    const wrapper = shallow(
      <Avatar label='boo' />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `label` without `name`', () => {
    const wrapper = shallow(
      <Avatar label='boo' />
    )
    expect(diveToLabel(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `label,name`', () => {
    const wrapper = shallow(
      <Avatar label='boo' name='bar' />
    )
    expect(diveToLabel(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `labelStyle`', () => {
    const wrapper = shallow(
      <Avatar labelStyle={{ backgroundColor: 'white' }} />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `labelTextStyle`', () => {
    const wrapper = shallow(
      <Avatar labelTextStyle={{ color: 'white' }} />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `nameStyle`', () => {
    const wrapper = shallow(
      <Avatar nameStyle={{ backgroundColor: 'white' }} />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `nameTextStyle`', () => {
    const wrapper = shallow(
      <Avatar nameTextStyle={{ color: 'white' }} />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `size:small`', () => {
    const wrapper = shallow(
      <Avatar size='small' />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })
})
