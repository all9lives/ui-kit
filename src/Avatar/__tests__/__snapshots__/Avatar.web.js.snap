// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar.web renders a snapshot 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={0}
  className="c0"
  direction="column"
  margin={0}
  padding={0}
  size="regular"
  source={1}
>
  <mapProps(Styled(Image))
    size={72}
    source={1}
    style={
      Object {
        "borderRadius": 0,
      }
    }
  />
</Styled(View)>
`;

exports[`Avatar.web renders a snapshot with \`borderRadius:5\` 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={5}
  className="c0"
  direction="column"
  margin={0}
  padding={0}
  size="regular"
  source={1}
>
  <mapProps(Styled(Image))
    size={72}
    source={1}
    style={
      Object {
        "borderRadius": 5,
      }
    }
  />
</Styled(View)>
`;

exports[`Avatar.web renders a snapshot with \`label,name\` 1`] = `
<View
  align=""
  direction="column"
  innerRef={[Function]}
  margin={0}
  name="bar"
  padding={0}
  style={
    Array [
      Object {
        "backgroundColor": "#ea4949",
        "borderRadius": 3,
        "bottom": 20,
        "flexDirection": "column",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 3,
        "paddingLeft": 3,
        "paddingRight": 3,
        "paddingTop": 3,
        "position": "absolute",
        "right": 1,
      },
      undefined,
    ]
  }
>
  <Styled(Text)
    alignSelf="auto"
    color="#191919"
    margin={0}
    size={16}
  >
    boo
  </Styled(Text)>
</View>
`;

exports[`Avatar.web renders a snapshot with \`label:boo\` 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={0}
  className="c0"
  direction="column"
  label="boo"
  margin={0}
  padding={0}
  size="regular"
  source={1}
>
  <mapProps(Styled(Image))
    size={72}
    source={1}
    style={
      Object {
        "borderRadius": 0,
      }
    }
  />
  <Styled(View)
    align=""
    direction="column"
    margin={0}
    padding={0}
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      boo
    </Styled(Text)>
  </Styled(View)>
</Styled(View)>
`;

exports[`Avatar.web renders a snapshot with \`label\` without \`name\` 1`] = `
<View
  align=""
  direction="column"
  innerRef={[Function]}
  margin={0}
  padding={0}
  style={
    Array [
      Object {
        "backgroundColor": "#ea4949",
        "borderRadius": 3,
        "bottom": 1,
        "flexDirection": "column",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 3,
        "paddingLeft": 3,
        "paddingRight": 3,
        "paddingTop": 3,
        "position": "absolute",
        "right": 1,
      },
      undefined,
    ]
  }
>
  <Styled(Text)
    alignSelf="auto"
    color="#191919"
    margin={0}
    size={16}
  >
    boo
  </Styled(Text)>
</View>
`;

exports[`Avatar.web renders a snapshot with \`labelStyle\` 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={0}
  className="c0"
  direction="column"
  labelStyle={
    Object {
      "backgroundColor": "white",
    }
  }
  margin={0}
  padding={0}
  size="regular"
  source={1}
>
  <mapProps(Styled(Image))
    size={72}
    source={1}
    style={
      Object {
        "borderRadius": 0,
      }
    }
  />
</Styled(View)>
`;

exports[`Avatar.web renders a snapshot with \`labelTextStyle\` 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={0}
  className="c0"
  direction="column"
  labelTextStyle={
    Object {
      "color": "white",
    }
  }
  margin={0}
  padding={0}
  size="regular"
  source={1}
>
  <mapProps(Styled(Image))
    size={72}
    source={1}
    style={
      Object {
        "borderRadius": 0,
      }
    }
  />
</Styled(View)>
`;

exports[`Avatar.web renders a snapshot with \`name:foo\` 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={0}
  className="c0"
  direction="column"
  margin={0}
  name="foo"
  padding={0}
  size="regular"
  source={1}
>
  <mapProps(Styled(Image))
    size={72}
    source={1}
    style={
      Object {
        "borderRadius": 0,
      }
    }
  />
  <Styled(View)
    align=""
    direction="column"
    margin={0}
    padding={0}
  >
    <Styled(Text)
      alignSelf="auto"
      color="#191919"
      margin={0}
      size={16}
    >
      foo
    </Styled(Text)>
  </Styled(View)>
</Styled(View)>
`;

exports[`Avatar.web renders a snapshot with \`nameStyle\` 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={0}
  className="c0"
  direction="column"
  margin={0}
  nameStyle={
    Object {
      "backgroundColor": "white",
    }
  }
  padding={0}
  size="regular"
  source={1}
>
  <mapProps(Styled(Image))
    size={72}
    source={1}
    style={
      Object {
        "borderRadius": 0,
      }
    }
  />
</Styled(View)>
`;

exports[`Avatar.web renders a snapshot with \`nameTextStyle\` 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={0}
  className="c0"
  direction="column"
  margin={0}
  nameTextStyle={
    Object {
      "color": "white",
    }
  }
  padding={0}
  size="regular"
  source={1}
>
  <mapProps(Styled(Image))
    size={72}
    source={1}
    style={
      Object {
        "borderRadius": 0,
      }
    }
  />
</Styled(View)>
`;

exports[`Avatar.web renders a snapshot with \`size:small\` 1`] = `
.c0 {
  position: relative;
}

<Styled(View)
  align=""
  borderRadius={0}
  className="c0"
  direction="column"
  margin={0}
  padding={0}
  size="small"
  source={1}
>
  <mapProps(Styled(Image))
    size={48}
    source={1}
    style={
      Object {
        "borderRadius": 0,
      }
    }
  />
</Styled(View)>
`;
